### DISCUSSION TITLE
Stub exported functions?

### DISCUSSION THREAD

**Discussion by theKidOfArcrania** [NVIDIA/open-gpu-kernel-modules#213] [Q&A]
There seems to be a number of stubbed out functions under `src/nvidia/arch/nvalloc/unix/src/exports-stubs.c` that are also exported to the final `nvidia.o_binary` object. Why is that the case? Are these functionalities just not open-sourced yet?

**Comment by YusufKhan-gamedev** [comment]
> There seems to be a number of stubbed out functions under `src/nvidia/arch/nvalloc/unix/src/exports-stubs.c` that are also exported to the final `nvidia.o_binary` object. Why is that the case? Are these functionalities just not open-sourced yet?

My assumption was that they are for things that are not supported on certain devices and for things that arent supported at all on the device.

**Reply by aaronp24** [reply]
These are stubs for features that are not supported by this version of the driver. [Chapter 44. Open Linux Kernel Modules](https://download.nvidia.com/XFree86/Linux-x86_64/515.43.04/README/kernel_open.html) in the README mentions some of these.
